Userland/Libraries/LibC/string.cpp

```diff
@@ -136,29 +136,26 @@ void* memcpy(void* dest_ptr, void const* src_ptr, size_t n)
         "rep movsb"
         : "+D"(dest_ptr), "+S"(src_ptr), "+c"(n)::"memory");
     return original_dest;
-#elif ARCH(AARCH64)
-    (void)dest_ptr;
-    (void)src_ptr;
-    (void)n;
-    TODO_AARCH64();
 #else
-#    error Unknown architecture
+    u8* pd = (u8*)dest_ptr;
+    u8 const* ps = (u8 const*)src_ptr;
+    for (; n--;)
+        *pd++ = *ps++;
+    return dest_ptr;
```

Comment:
I wonder if compilers are smart enough to see that this is memcpy(), then transform it to a call to `__builtin_memset()` internally, which can then generate a call to memcpy() under certain circumstances. Do we use any flags or labels.

Well, if it becomes a problem, we can compile this file with `-fno-builtin-memset -fno-builtin-memcpy` I suppose.

(Or maybe my mental model on how this works is incomplete – well possible.)

Reply:
We add these flags for the LibC build, so I think it should be fine (at least for the GCC build).
https://github.com/SerenityOS/serenity/blob/84e8d5f41842f6c3f9f613f3570f669328fe4d06/Userland/Libraries/LibC/CMakeLists.txt#L138-L141